### PR TITLE
[MNG-6294] Convert MavenPluginValidator into a plexus component

### PR DESCRIPTION
**Pull request by msimacek**
Patch for https://issues.apache.org/jira/browse/MNG-6294

### CODE REVIEW

File: maven-core/src/test/java/org/apache/maven/plugin/MavenPluginValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    public void testInvalid()
+    {
+        Artifact plugin = new DefaultArtifact( "org.apache.maven.its.plugins", "maven-it-plugin", "0.1", "compile",
+                "jar", null, new DefaultArtifactHandler() );
+        PluginDescriptor descriptor = new PluginDescriptor();
+        descriptor.setGroupId( "org.apache.maven.its.plugins" );
+        descriptor.setArtifactId( "maven-it-plugin" );
+        List<String> errors = new ArrayList<>();
+        mavenPluginValidator.validate( plugin, descriptor, errors );
+        assertFalse( errors.isEmpty() );
```
Comment (eolivelli) [inline maven-core/src/test/java/org/apache/maven/plugin/MavenPluginValidatorTest.java]: nit: we should test more the "errors" result, it is not clear that we are catching the error we are generating with the lines above.
We have three validations which can fire an error, we should test every of them.

**Comment by michael-o** [comment]
Willing to merge if someone can extend the tests.

**Comment by msimacek** [comment]
Hi, I currently don't have time to continue with this PR, but @mizdebsk will take it over and complete the tests. Please be patient.

**Comment by mizdebsk** [comment]
Ack, I will take over this PR from @msimacek - rebase it and extend unit tests.

**Comment by michael-o** [comment]
Thanks guys, no need to rush. Will just wait.

**Comment by rfscholte** [comment]
hmm, we need to rethink this a bit, I'm not sure if passing a list to get the errors is the preferred way. Also I would consider to use a Problem or something similar instead, so we can provide more detailed information.

**Pull request closed without merging**